### PR TITLE
Renode expect support

### LINKED ISSUE #271: Renode: there should be a single-command to test in Renode
When I test on arty, I can run

```
$ make unit
```

Which will load the C code, choose some menu items then exit back to the command line.

It would be nice if there were an equivalent with renode, as it would allow faster iteration. 

I seem to recall that there used to be something like this for renode? Or am I misremembering?

### PR DESCRIPTION
**Pull request by PiotrZierhoffer**
This adds 3 targets:

- renode-headless, which just runs Renode in a single console
- unit-renode, same as `make unit` but running in Renode
- run-renode, similar to `make run`, added for completeness

All of these accept the `TARGET` variable to select the actual target.

Closes #271

### CODE REVIEW

File: proj/proj.mk [hunk 1]
```diff
@@ -273,6 +277,14 @@ unit: $(SOFTWARE_BIN)
 	@echo Running unit test on board
 	$(BUILD_DIR)/interact.expect $(SOFTWARE_BIN) $(TTY) $(UART_SPEED) $(TEST_MENU_ITEMS) |& tee $(UNITTEST_LOG)
 
+run-renode: $(SOFTWARE_ELF) renode-scripts
+	@echo Running automated test in Renode
+	$(BUILD_DIR)/interact.expect r $(RUN_MENU_ITEMS) |& tee $(UNITTEST_LOG)
```
Comment (tcal-x) [inline proj/proj.mk:282]: Should be $(SOFTWARE_LOG)

Comment (PiotrZierhoffer) [reply proj/proj.mk]: Good catch, fixed

**Review by tcal-x — APPROVED** [review]
LGTM

**Comment by alanvgreen** [comment]
Thanks!

**Pull request merged**